source/posix/system_info.c

```diff
@@ -109,6 +109,41 @@ struct aws_stack_frame_info {
     char function[128];
 };
 
+/* strip shell special characters from the exe in case someone tries to
+   rename the exe and trigger shell execution via the sub commands used to
+   resolve symbols */
+char *s_strip_shell_chars(char *path) {
+    char *cur = path;
+    while (*cur) {
+        switch (*cur) {
+            case '\\':
+            case '>':
+            case '<':
+            case '$':
+            case ';':
+            case '|':
+            case '[':
+            case ']':
+            case '`':
+            case '(':
+            case ')':
+            case '&':
+            case '#':
+            case '@':
+            case '!':
+            case '^':
+            case '=':
+            case '?':
+                *cur = '_';
```

Comment:
Why not `if (!(isalpha(*cur) || isspace(*cur))) { *cur = '_'; }`

Reply:
/path/to/do-the-thing.exe

No really, I did think about doing this as a whitelist, because that's the correct thing to do, but it's restrictive because it does limit what a customer can put in their binary name. I'm happy to do it either way.